plutus-core/untyped-plutus-core-test/Evaluation/Golden.hs

```diff
@@ -313,10 +314,12 @@ goldenVsPretty extn name value =
         either id (BSL.fromStrict . encodeUtf8 . render . prettyPlcClassicDebug) <$> runExceptT value
 
 goldenVsEvaluatedCK :: String -> Term TyName Name DefaultUni DefaultFun () -> TestTree
-goldenVsEvaluatedCK name = goldenVsPretty ".plc.golden" name . pure . evaluateCk defBuiltinsRuntime
+goldenVsEvaluatedCK name =
+    goldenVsPretty ".plc.golden" name . pure . fmap UPLC.erase . evaluateCk defBuiltinsRuntime
 
 goldenVsEvaluatedCEK :: String -> Term TyName Name DefaultUni DefaultFun () -> TestTree
-goldenVsEvaluatedCEK name = goldenVsPretty ".plc.golden" name . pure . evaluateCek defBuiltinsRuntime
+goldenVsEvaluatedCEK name =
+    goldenVsPretty ".plc.golden" name . pure . evaluateCek defBuiltinsRuntime . UPLC.erase
```

Comment:
Output is untyped now.

Reply:
Do we really want to do that?  Conceivably there could be a bug in, for example, the type-substitution code in the CK machine that would produce ill-typed code that looks OK after erasure.  Maybe that kind of thing is being tested elsewhere? We have a lot of tests these days and I've lost track a bit.

Reply:
Type substitution is hard to test - I am generally in favour more tests that can catch this.

Reply:
> Conceivably there could be a bug in, for example, the type-substitution code in the CK machine that would produce ill-typed code that looks OK after erasure.

Added a test for that in [this](https://github.com/input-output-hk/plutus/pull/2744/commits/bc6bae6a3cb693f0b111df0bdfaf25ba0b40edf3) commit.

Reply:
great!